kernel/src/rsstart.rs

```diff
@@ -17,8 +17,11 @@ use crate::{
     vm,
 };
 
+use common::refcnt::RefCnt;
+use common::ringbuffer::{RingBuffer, RingBufferRawData};
+
 extern "C" {
-    fn kmain();
+    fn kmain(rb_in: RefCnt<RingBuffer>, rb_out: RefCnt<RingBuffer>);
```

Comment:
I don't think kmain should take these as parameters, there should be a global static `RingBuffer` that's set up by `rsstart`. Also why is it `RefCnt`ed? The outside can allocate it `Box`ed and then leak it or ensure it's only freed after the vm exits (like with the allocator), so it'll live "forever" without a reference count anyway.

Basically the logic here should match the BuddyAllocator.